locking.go

```diff
@@ -137,28 +135,50 @@ var (
 // NewLockInstance returns a new instance of a lock
-func NewLockInstance(ctxParent context.Context, accountName, accountKey, lockName string, lockTTL time.Duration, behavior ...BehaviorFunc) (*Lock, error) {
-	if accountKey == "" {
+//
+// Params
+// StorageAccountURL: HTTPS endpoint for your storage account eg. `https://mystorageaccount.blob.core.windows.net` if your account is named `mystorageaccount`
+// StorageAccountKey: The access key for your storage account
+// LockName: An alphanumberic string < 58 chars that will represent your lock.
+// LockTTL: A duration between 15 and 60 seconds for which the lock will be held. Note, by default the `AutoRenew` behavior will renew locks until `Unlock` is called
+//
+// Advanced
+// Behaviors: Funcs which allow you to mutate the lockInstance's behavior. Leave empty for default behavior
+//
+func NewLockInstance(ctxParent context.Context, storageAccountURL, storageAccountKey, lockName string, lockTTL time.Duration, behavior ...BehaviorFunc) (*Lock, error) {
+	if storageAccountKey == "" {
 		return nil, fmt.Errorf("Empty accountKey is invalid")
 	}
-	if accountName == "" {
-		return nil, fmt.Errorf("Empty accountName is invalid")
-	}
 	if lockTTL.Seconds() < 15 || lockTTL.Seconds() > 60 {
 		return nil, fmt.Errorf("LockTTL of %v seconds is outside allowed range of 15-60seconds", lockTTL.Seconds())
 	}
 
-	creds := azblob.NewSharedKeyCredential(accountName, accountKey)
+	storageAccountURLParsed, err := url.Parse(storageAccountURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse storageAccountUrl, err: %+v", err)
+	}
+	if storageAccountURLParsed.Scheme != "https" {
+		return nil, fmt.Errorf("storageAccountURL should be 'https' eg. 'https://mystorageaccount.blob.core.windows.net'")
+	}
+	// Extract the accountname from the storage URL
+	// for example 'https://mystorageaccount.blob.core.windows.net' -> 'mystorageaccount'
+	accountName, err := extractAccountNameFromURL(storageAccountURLParsed)
+	if err != nil {
+		return nil, err
+	}
+
+	creds := azblob.NewSharedKeyCredential(accountName, storageAccountKey)
 
 	// Create a ContainerURL object to a container
-	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, lockName))
+	u, _ := url.Parse(fmt.Sprintf("%s/%s", storageAccountURL, lockName))
 	containerURL := azblob.NewContainerURL(*u, azblob.NewPipeline(creds, azblob.PipelineOptions{}))
-	_, err := containerURL.Create(ctxParent, nil, azblob.PublicAccessNone)
+	_, err = containerURL.Create(ctxParent, nil, azblob.PublicAccessNone)
 
 	// Create will return a '409' response code if the container already exists
 	// we only error on other conditions as it's expected that a lock of this
 	// name may already exist
-	if err != nil && err.(azblob.ResponseError) != nil && err.(azblob.ResponseError).Response().StatusCode != 409 {
+	_, isReponseError := err.(azblob.ResponseError)
+	if err != nil && isReponseError && err.(azblob.ResponseError).Response().StatusCode != 409 {
```

Comment:
Why do the type assertion twice and not just use the result of the line above here?

Reply:
Merge fail :( Will fix